### PR TITLE
Filter cases in past x days

### PR DESCRIPTION
**Pull request by seallard**
## Description
Replace call to gigantic and complex
```python
cases: List[Family] = db.cases(days=31)
```
With
```python
cases: List[Family] = db.get_cases_created_within_days(days=31)
```

The goal is to reduce the number of places the cases function is used in order to be able to simplify and refactor it more easily.

### Added
- get_cases_in_past_days(days: int) -> List[Family]
- GET_NEW filter

### Changed
- Replaced call to cases with get_cases_in_past_days(days=31)
- Rename GET_OLD to GET_NEW

## Review
- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/store/filters/status_case_filters.py [hunk 1]
```diff
@@ -213,7 +219,8 @@ class CaseFilter(Enum):
 
     GET_HAS_SEQUENCE: Callable = get_cases_has_sequence
     GET_HAS_INACTIVE_ANALYSIS: Callable = get_inactive_analysis_cases
-    GET_NEW: Callable = get_new_cases
+    GET_OLD: Callable = get_older_cases
+    GET_NEW: callable = get_newer_cases
```
Comment (Vince-janv) [inline cg/store/filters/status_case_filters.py:223]:
```suggestion
    GET_NEW: Callable = get_newer_cases
```

File: tests/store/api/find/test_find_business_data.py [hunk 2]
```diff
@@ -731,3 +731,32 @@ def test_get_case_by_name_and_customer_case_found(store_with_multiple_cases_and_
     assert filtered_case is not None
     assert filtered_case.customer_id == customer.id
     assert filtered_case.name == case.name
+
+
+def test_fetch_cases_newer_than_date_no_cases(store_with_multiple_cases_and_samples: Store):
+    """Test that no cases are returned when there are no cases newer than the given date."""
+    # GIVEN a store with cases older than 7 days
+    older_than_date = datetime.now() - timedelta(days=10)
+    for case in store_with_multiple_cases_and_samples._get_query(table=Family).all():
```
Comment (Vince-janv) [inline tests/store/api/find/test_find_business_data.py:740]:
```suggestion
    for case in store_with_multiple_cases_and_samples._get_query(table=Family):
```

File: tests/store/api/find/test_find_business_data.py [hunk 3]
```diff
@@ -731,3 +731,32 @@ def test_get_case_by_name_and_customer_case_found(store_with_multiple_cases_and_
     assert filtered_case is not None
     assert filtered_case.customer_id == customer.id
     assert filtered_case.name == case.name
+
+
+def test_fetch_cases_newer_than_date_no_cases(store_with_multiple_cases_and_samples: Store):
```
Comment (Vince-janv) [inline tests/store/api/find/test_find_business_data.py:736]: Super clean tests 🌟

**Review by Vince-janv — APPROVED** [review]
Good idea to reduce the usage of `.cases()` 👍

File: cg/store/api/status.py [hunk 4]
```diff
@@ -233,7 +233,7 @@ def get_cases_to_compress(self, date_threshold: datetime) -> List[Family]:
         """Return all cases that are ready to be compressed by SPRING."""
```
Comment (diitaz93) [inline cg/store/api/status.py:233]:
```suggestion
        """Return all cases that are ready to be compressed by SPRING created before a given date."""
```
Also maybe change the parameter name from `date_threshold` to `before_date` or something more meaningful

File: cg/store/filters/status_case_filters.py [hunk 5]
```diff
@@ -35,12 +35,18 @@ def get_running_cases(cases: Query, **kwargs) -> Query:
     return cases.filter(Family.action == CaseActions.RUNNING)
 
 
-def get_new_cases(cases: Query, date: datetime, **kwargs) -> Query:
-    """Return old cases compared to date."""
+def get_older_cases(cases: Query, date: datetime, **kwargs) -> Query:
+    """Return older cases compared to date."""
```
Comment (diitaz93) [inline cg/store/filters/status_case_filters.py:39]:
```suggestion
    """Return cases created before a certain date."""
```

File: cg/store/filters/status_case_filters.py [hunk 6]
```diff
@@ -35,12 +35,18 @@ def get_running_cases(cases: Query, **kwargs) -> Query:
...
     cases = cases.filter(Family.created_at < date)
     return cases.order_by(Family.created_at.asc())
 
 
+def get_newer_cases(cases: Query, date: datetime, **kwargs) -> Query:
+    """Return newer cases compared to date."""
```
Comment (diitaz93) [inline cg/store/filters/status_case_filters.py:45]:
```suggestion
    """Return cases created after a certain date."""
```

File: tests/store/api/find/test_find_business_data.py [hunk 7]
```diff
@@ -1,6 +1,6 @@
 """Tests the findbusinessdata part of the Cg store API."""
```
Comment (diitaz93) [inline tests/store/api/find/test_find_business_data.py:1]:
```suggestion
"""Tests the find business data part of the Cg store API."""
```

**Pull request merged**